mantis-runtime/src/main/java/io/mantisrx/runtime/scheduler/MantisRxScheduler.java

```diff
@@ -16,25 +16,28 @@
 
 package io.mantisrx.runtime.scheduler;
 
+import static java.util.concurrent.Executors.newFixedThreadPool;
+
+import java.util.concurrent.ExecutorService;
 import java.util.concurrent.ThreadFactory;
 
 import rx.Scheduler;
-import rx.internal.schedulers.NewThreadWorker;
-
+import rx.schedulers.Schedulers;
 
 /**
- * Schedules work on the same thread that is constructed once as part of the creating this Scheduler.
+ * Schedules work on the same fixed thread pool executor with 1 thread that is constructed once as part of creating
+ * this Scheduler.
  */
-public final class SingleThreadScheduler extends Scheduler {
-
-    private final NewThreadWorker worker;
+public final class MantisRxScheduler extends Scheduler {
```

Comment:
Maybe a nitpick, should we name it to signify this is a single threaded scheduler?

Reply:
I went back and forth on that, can revert the naming if that is more clear.

Reply:
updated to MantisRxSingleThreadScheduler